File: Maximize_the_Confusion_of_an_Exam.py
class Solution:
    def maxConsecutiveAnswers(self, answerKey: str, k: int) -> int:
        window_size = 0
        count = {"T":0, "F":0}

        for right in range(len(answerKey)):
            count[answerKey[right]] += 1
            minor = min(count["T"], count["F"])

            if minor <= k:
                window_size += 1
            else:
                count[answerKey[right - window_size]] -= 1
        
        return window_size

File: Maximize_the_Confusion_of_an_Exam.js
/**
 * @param {string} answerKey
 * @param {number} k
 * @return {number}
 */
var maxConsecutiveAnswers = function(answerKey, k) {
    let windowSize = 0;
    let count = { T: 0, F: 0 };

    for (let right = 0; right < answerKey.length; right++) {
        count[answerKey[right]] += 1;
        let minor = Math.min(count["T"], count["F"]);

        if (minor <= k) {
            windowSize += 1;
        } else {
            count[answerKey[right - windowSize]] -= 1;
        }
    }

    return windowSize;    
};
